tests/specs/stn/with-replay-error-mode.e2e.js

```diff
@@ -0,0 +1,152 @@
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.waitForAgentLoad()
+    await browser.waitUntil(() => browser.execute(async function () { // here for no-retry wdio stability
+      return await Object.values(newrelic.initializedAgents)[0].features.session_trace.onAggregateImported
+    }), { timeout: 5000 })
+    let startingMode = (await getTraceMode())[0]
+    expect(initSTReceived).toBeUndefined()
+    expect(startingMode).toEqual(MODE.ERROR)
+
+    await browser.execute(simulateErrorInBrowser)
+    await getFirstSTPayload
+    let modeAfterErr = (await getTraceMode())[0]
+    expect(modeAfterErr).toEqual(MODE.FULL)
+    // The loadEventEnd is part of PT and PNT resource entry and is a node created at start of page life.
+    expect(initSTReceived.request.body.res.find(node => node.n === 'loadEventEnd')).toEqual(expect.objectContaining({ n: 'loadEventEnd', o: 'document', t: 'timing' }))
+
+    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy() // double check there's nothing wrong with full mdoe interval harvest
+  })
+
+  it('starts in full mode when an error happens before page load', async () => {
+    await browser.url(await getReplayOnErrorUrl)
+    await browser.execute(simulateErrorInBrowser)
+    await browser.waitForAgentLoad()
+    await expect(getFirstSTPayload).resolves.toBeTruthy()
+    let startingMode = (await getTraceMode())[0]
+    expect(startingMode).toEqual(MODE.FULL)
+
+    await expect(browser.testHandle.expectResources(3000)).resolves.toBeTruthy()
+  })
+
+  it.withBrowsersMatching(onlyChrome)('does not capture more than the last 30 seconds when error happens', async () => {
+    await getReplayOnErrorUrl.then(builtUrl => browser.url(builtUrl)).then(() => browser.waitForAgentLoad()).then(async () => {
+      await browser.pause(30000)
+      await Promise.all([
+        browser.testHandle.expectResources(3000), // the setup expectRes promise would've timed out already -- you will see some console errors but they don't impact validity
+        browser.execute(simulateErrorInBrowser)
+      ]).then(async ([initSTAfterErr]) => {
+        expect(initSTAfterErr.request.body.res.find(node => node.n === 'loadEventEnd')).toBeUndefined() // that node should've been tossed out by now
+      })
+    })
+  })
+
+  it('does not perform final harvest while in this mode', async () => {
+    await getReplayOnErrorUrl.then(builtUrl => browser.url(builtUrl)).then(() => browser.waitForAgentLoad()).then(async () => {
+      await getTraceMode().then(traceState => expect(traceState).toEqual([MODE.ERROR, false])) // sanity check tbh
+      expect(initSTReceived).toBeUndefined()
+    }).then(() => browser.refresh()).then(() => browser.waitForAgentLoad()).then(async () => {
+      await getTraceMode().then(traceState => expect(traceState).toEqual([MODE.ERROR, false]))
+      expect(initSTReceived).toBeUndefined() // no harvest from either previous unload or from new existing-session load
```

Comment:
Because of the way this is set up, Is there a chance that the STN payload could have sent -- but latency causes  this expect to occur before properly waiting for the testing server to get it? Would it be safer to just wait for the resources call with a timeout?

Reply:
Patrick kind of brought this up earlier in week; could change `waitForAgentLoad` to wait for the feature load, whatever that command is, instead